packages/controller-utils/src/util.ts

```diff
@@ -120,7 +120,7 @@ export function getBuyURL(
  * @returns A BN instance.
  */
 export function hexToBN(inputHex: string) {
-  return new BN(stripHexPrefix(inputHex), 16);
+  return new BN(inputHex ? stripHexPrefix(inputHex) : inputHex, 16);
```

Comment:
This was required because a bunch of tests started failing since `ethereumjs-util` version is dedupe bumped to `7.1.5`, which changes the `stripHexPrefix` function [such that it throws when not a string](https://npmfs.com/package/ethereumjs-util/7.1.5/src/internal.ts#L46), where before it would simple return the input when not a string.

Reply:
Makes sense, but can this change go in another PR? Where is this being called?

Reply:
So the failures are happening on this line: https://github.com/MetaMask/core/blob/782004d2d1d10187b1afde680e4c35be7f8335ce/packages/transaction-controller/src/TransactionController.ts#L1010 in a suite of `TransactionsController` tests that aren't touched here at all. Its all just from the version deduping happening here: https://github.com/MetaMask/core/pull/1064/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deR4753

Reply:
same as here: https://github.com/MetaMask/core/pull/1064#discussion_r1139424170.

This change has been reverted